.github/workflows/sanity-checks.yml

```diff
@@ -0,0 +1,86 @@
+on:
+  # push:
+  #   branches: [main, 'release-v**']
+  pull_request:
+    paths:
+      - 'src/'
```

Comment:
This was my attempt to hint the questions "What should trigger a battery of tests?"
Knowing these would allow for the pipeline to trigger this massive checks only when necessary. For instance, if you update the `scripts` or the `LICENSE` or `README` file
But honestly now that I'm more familiar with the code I have a feeling there's very little of this cases where modifying a file does not require to run all sanity checks, thus unless you guys tell me otherwise let's run it always

Reply:
I think we could exclude changes to any file type which is neither `rs`, nor `toml` nor `sh`.

Reply:
Yeah, I would say 95% of PR requires sanity checks. If we can add the William patterns, then ok. If not, I think right now, going ahead with everything is fine